### ISSUE TITLE
Cannot start Kafka client for cluster local: kafka: client has run out of available brokers to talk to (Is your cluster reachable?)

### ISSUE THREAD

**Issue by bencoughlan** [linkedin/Burrow#226]
Can't connect to brokers, even though i can curl, etc. 

Any idea why this would be? 

burrow.cfg

[general]
logconfig=/etc/burrow/logging.cfg
group-blacklist=^(console-consumer-|python-kafka-consumer-).*$

[zookeeper]
hostname=zookeeper00.kafkauat.com 
hostname=zookeeper01.kafkauat.com 
hostname=zookeeper02.kafkauat.com 
port=2181
timeout=6
lock-path=/burrow/notifier

[kafka "local"]
broker=broker00.kafkauat.com 
broker=broker01.kafkauat.com
broker=broker02.kafkauat.com
broker-port=9092
zookeeper=zookeeper00.kafkauat.uat.aws.ad.zopa.com
zookeeper-path=/
zookeeper-offsets=true
offsets-topic=__consumer_offsets

[tickers]
broker-offsets=60

[lagcheck]
intervals=10
expire-group=604800

[httpserver]
server=on
port=8000

log output : 
2017-06-07 18:05:35 [INFO] Starting Zookeeper client
2017-06-07 18:05:35 [INFO] Starting Offsets Storage module
2017-06-07 18:05:35 [INFO] Starting HTTP server
2017-06-07 18:05:35 [INFO] Starting Zookeeper client for cluster local
2017-06-07 18:05:35 [INFO] Starting Kafka client for cluster local
2017-06-07 18:05:35 [CRITICAL] Cannot start Kafka client for cluster local: kafka: client has run out of available brokers to talk to (Is your cluster reachable?)

**Comment by joadha** [comment]
Same issue here. The cluster is definitely reachable from the machine on which I'm running Burrow. Frustrating error.

**Comment by toddpalino** [comment]
The error is simply that Burrow cannot connect to any of the 3 brokers specified on port 9092. This could be due to a connection problem (bad hostname, bad port, no network access, firewall, etc.). It could also be because the Kafka cluster is rejecting the connection for some reason (you're not using TLS, but a bad TLS handshake or certificate validation would be something like that).

**Comment by toddpalino** [comment]
Closing this out. Please ask on #burrow in the Confluent Community if you continue to have issues and need some help with configs.

**Comment by mikolajmorawski** [comment]
Thanks for your help, after this change it started to work correctly. I was trying all possible configurations but I haven’t thought about adding field handshake-first=true to my configuration to override this “default” value :D, so it was a problem with the wrong default value :D

The next step I was doing was to connect Burrow with SASL_SSL secured kafka. In documentation there is the note about certfile and keyfile filed which are required. It is not true when you want to connect to SSL secured kafka. The only thing I had to do was to set this:

```
[tls.mytls]
noverify=true
```

I found that there is a mistake in documentation, the field there is shown as “no-verify” but correct value is “noverify” like in your code. 
The problem I had with this that it was not possible to debug. The Burrow logs only shows this debug message: 

`“Cannot start Kafka client for cluster local: kafka: client has run out of available brokers to talk to (Is your cluster reachable?)”`

This message could be related to any problem, by I already knew that the last thing that was not working was TLS. So changing this field name helped me. 
Also this other fields are not required during SSL connection to Kafka. Maybe there are required when you are creating HttpServer with TLS. It would be nice to split this two things and describe them separately.
